dot/node_test.go

```diff
@@ -0,0 +1,437 @@
+				gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(&grandpa.Service{}, nil)
+			m.EXPECT().newSyncService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}), &grandpa.Service{},
+				&babe.VerificationManager{}, &core.Service{}, gomock.AssignableToTypeOf(&network.Service{}),
+				gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(&gsync.Service{}, nil)
+			m.EXPECT().createBABEService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}), ks.Babe,
+				&core.Service{}, gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(&babe.Service{}, nil)
+			m.EXPECT().createSystemService(&tt.args.cfg.System, gomock.AssignableToTypeOf(&state.Service{})).
+				DoAndReturn(func(cfg *types.SystemInfo, stateSrvc *state.Service) (*system.Service, error) {
+					gd, err := stateSrvc.Base.LoadGenesisData()
+					return system.NewService(cfg, gd), err
+				})
+			m.EXPECT().createNetworkService(tt.args.cfg, gomock.AssignableToTypeOf(&state.Service{}),
+				gomock.AssignableToTypeOf(&telemetry.Mailer{})).Return(testNetwork, nil)
+
+			got, err := newNode(tt.args.cfg, ks, m)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.want != nil {
+				assert.NotNil(t, got)
+			} else {
+				assert.Nil(t, got)
+			}
+		})
+	}
+}
+
+func createTestService(t *testing.T, cfg *network.Config) (srvc *network.Service) {
+	t.Helper()
+	ctrl := gomock.NewController(t)
+
+	if cfg == nil {
+		cfg = &network.Config{
+			BasePath:     t.TempDir(),
+			NoBootstrap:  true,
+			NoMDNS:       true,
+			LogLvl:       log.Warn,
+			SlotDuration: time.Second,
+		}
+	}
+	if cfg.BlockState == nil {
+		blockstate := network.NewMockBlockState(ctrl)
+
+		cfg.BlockState = blockstate
+	}
+
+	cfg.SlotDuration = time.Second
+
+	if cfg.Telemetry == nil {
+		telemetryMock := NewMockClient(ctrl)
+		telemetryMock.EXPECT().SendMessage(gomock.Any()).AnyTimes()
+		cfg.Telemetry = telemetryMock
+	}
+
+	srvc, err := network.NewService(cfg)
+	require.NoError(t, err)
+
+	return srvc
+}
+
+func TestNodeInitialized(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := newTestGenesisRawFile(t, cfg)
+	require.NotNil(t, genFile)
+
+	cfg.Init.Genesis = genFile
+
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	type args struct {
+		basepath string
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "blank base path",
+			args: args{basepath: ""},
+			want: false,
+		},
+		{
+			name: "working example",
+			args: args{basepath: cfg.Global.BasePath},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NodeInitialized(tt.args.basepath); got != tt.want {
+				t.Errorf("NodeInitialized() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func initKeystore(cfg *Config) (*keystore.GlobalKeystore, error) {
```

Comment:
Is this used anywhere outside tests? If not, pass in a `*testing.T` and use `require.NoError(t, err)` for each possible error.

Reply:
Updated.